internal/database/users.go

```diff
@@ -1191,7 +1191,21 @@ func (u *userStore) getOneBySQL(ctx context.Context, q *sqlf.Query) (*types.User
 
 // getBySQL returns users matching the SQL query, if any exist.
 func (u *userStore) getBySQL(ctx context.Context, query *sqlf.Query) ([]*types.User, error) {
-	q := sqlf.Sprintf("SELECT u.id, u.username, u.display_name, u.avatar_url, u.created_at, u.updated_at, u.site_admin, u.passwd IS NOT NULL, u.tags, u.invalidated_sessions_at, u.tos_accepted, u.searchable FROM users u %s", query)
+	q := sqlf.Sprintf(`
+SELECT u.id,
+       u.username,
+       u.display_name,
+       u.avatar_url,
+       u.created_at,
+       u.updated_at,
+       u.site_admin,
+       u.passwd IS NOT NULL,
+       u.tags,
+       u.invalidated_sessions_at,
+       u.tos_accepted,
+       u.searchable,
+       (SELECT COUNT(user_id) FROM user_external_accounts WHERE user_id=u.id AND service_type = 'scim') >= 1 AS scim_controlled
+FROM users u %s`, query)
```

Comment:
Wondering if you can simplify this to not do a SQL query on every row, something like:

```suggestion
SELECT u.id,
       u.username,
       u.display_name,
       u.avatar_url,
       u.created_at,
       u.updated_at,
       u.site_admin,
       u.passwd IS NOT NULL,
       u.tags,
       u.invalidated_sessions_at,
       u.tos_accepted,
       u.searchable,
       COUNT(uea.service_type) > 0 AS scim_controlled
FROM users u
LEFT JOIN user_external_accounts uea
       ON u.id = uea.user_id
       AND uea.service_type = 'scim'
%s`, query)
```

Reply:
Fair point, I’ll take care of it in a follow-up PR!
















Message Petri Last








Shift + Return to add a new line
Thread
[ask-design](https://sourcegraph.slack.com/archives/C0HMGV90V/p1678214796557589)